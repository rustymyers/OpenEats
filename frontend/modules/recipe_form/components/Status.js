import React from 'react'

class Status extends React.Component {
  render() {
    if (this.props.status.message.length > 1) {
      let cssClass = "alert alert-info alert-dismissible";
      if (this.props.status.alert) {
        cssClass = "alert alert-dismissible " + this.props.status.alert;
      }
      return (
        <div className="row">
          <div className="col-xs-12">
            <div className={ cssClass } role="alert">
              <button type="button" className="close" data-dismiss="alert" onClick={ this.props.actions.close }>
                <span>&times;</span>
              </button>
              { this.props.status.message }
            </div>
          </div>
        </div>
      )
    }
    return (
      <span/>
    )
  }
}

export default Status;
