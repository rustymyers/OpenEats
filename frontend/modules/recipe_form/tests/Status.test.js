import React from 'react';
import Status from '../components/Status';
import renderer from 'react-test-renderer';

test('Status component test', () => {
  const component = renderer.create(
    <Status status={{message: []}}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
