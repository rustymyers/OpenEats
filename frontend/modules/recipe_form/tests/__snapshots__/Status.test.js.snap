// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component test 1`] = `<span />`;
